instructor/common.py

```diff
@@ -1097,6 +1100,88 @@ def facs_sample_prep(request):
         context_instance=RequestContext(request))
 
 
+def create_facs_histograms(assignment, facs_sample):
+    strain_protocols = models.StrainTreatment.objects.filter(assignment=assignment)
+    for strain_protocol in strain_protocols:
+        histogram, created = models.FlowCytometryHistogram.objects.get_or_create(
+            sample_prep=facs_sample,
+            strain_protocol=strain_protocol,
+        )
+
+
+@check_assignment_owner
+@login_required
+def facs_analyze(request):
+    page_number = page_order.index('facs_analyze')
+    pk = request.session['assignment_id']
+    assignment = models.Assignment.objects.get(id=pk)
+
+    HistogramFormset = modelformset_factory(
+        models.FlowCytometryHistogram,
+        extra=0,
+        exclude=['sample_prep', 'strain_protocol']
+    )
+    if request.method == 'POST' and assignment.access == 'private':
+        formset = HistogramFormset(request.POST)
+        if formset.is_valid():
+            entries = formset.save(commit=False)
+            for entry in entries:
+                entry.save()
+            if 'continue' in request.POST:
+                if assignment.last_enabled_page <= page_number:
+                    assignment.last_enabled_page = page_number+1
+                    assignment.save()
+                return redirect('common_assignments')
+    elif request.method == "POST" and 'continue' in request.POST:
+        return redirect('common_assignments')
```

Comment:
Nit: I would be this condition as the first of the 3, considering it has the simplest code (but feel free to leave as it is)